yt_dlp/extractor/imdb.py

```diff
@@ -73,11 +98,13 @@ def _real_extract(self, url):
             ext = mimetype2ext(encoding.get(
                 'mimeType')) or determine_ext(video_url)
             if ext == 'm3u8':
-                formats.extend(self._extract_m3u8_formats(
+                fmts, subs = self._extract_m3u8_formats_and_subtitles(
                     video_url, video_id, 'mp4', entry_protocol='m3u8_native',
-                    preference=1, m3u8_id='hls', fatal=False))
+                    preference=1, m3u8_id='hls', fatal=False)
+                subtitles = self._merge_subtitles(subtitles, subs)
+                formats.extend(fmts)
                 continue
-            format_id = encoding.get('definition')
+            format_id = try_get(encoding, lambda x: x['displayName']['value']) or encoding.get('definition')
```

Comment:
```suggestion
            format_id = traverse_obj(encoding, ('displayName', 'value'), 'definition')
```
All your uses of try_get can actually be replaced with traverse_obj, but for this specific case in purticular, it makes no sense to use try_get